### PR TITLE
DatePicker - DateInput - DateRangeInput

### LINKED ISSUE #51: Add DateInput
**Note**: we will likely want to include a two-panel `DateInput` as well.

-------

Normal:

image
-------------
Expanded:

image
------------
Range:

image

Location: `src/Form/DateInput.js`
Potential components: [react-datepicker](https://github.com/Hacker0x01/react-datepicker), [react-dates](https://github.com/airbnb/react-dates), [react-day-picker](https://github.com/gpbl/react-day-picker)

### PR DESCRIPTION
**Pull request by lmcorbalan**
Closes #51 
Depends on #265 

This PR adds 3 new components:
 - DatePicker
 - DateInput
 - DateRangeInput

Thanks to @sistemico for creating DatePicker and DateInput
Thanks to @sgobotta for implementing DateRangeInput

### CODE REVIEW

File: src/components/DatePicker/DatePicker.js [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+      </Container>
+    )
+  }
+}
+
+DatePicker.propTypes = {
+  currentDate: PropTypes.instanceOf(Date),
+
+  // Events
+  onSelect: PropTypes.func,
```
Comment (AquiGorka) [inline src/components/DatePicker/DatePicker.js]: We've been migrating `onSelect` -> `onChange`, could you update the components?

File: src/components/Input/DateInput.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+class DateInput extends React.PureComponent {
+  state = {
+    showPicker: false,
+  }
+
+  get formattedValue() {
+    const { value } = this.props
+
+    return isDate(value) ? formatDate(value, this.props.format) : ''
```
Comment (AquiGorka) [inline src/components/Input/DateInput.js]: As you are already destructuring `value` may I suggest:
```JavaScript
get formattedValue() {
  const { value, format } = this.props
  return isDate(value) ? formatDate(value, format) : ''
}
```

**Comment by luisivan** [comment]
@lmcorbalan sorry for the delay in getting this reviewed -- now we are 200% focused on the 0.7 release of the Aragon client, so we will have to delay this another couple of weeks. But we have scheduled time for  @bpierre to provide his review during the last week of April

**Comment by luisivan** [comment]
@lmcorbalan sorry for the delay in getting this reviewed -- now we are 200% focused on the 0.7 release of the Aragon client, so we will have to delay this another couple of weeks. But we have scheduled time for  @bpierre to provide his review during the last week of April

**Comment by sohkai** [comment]
Closing for https://github.com/aragon/aragon-ui/pull/482; sorry we couldn't pull this into aragonUI fast enough, we've been using and changing this component in https://github.com/aragon/aragon-apps over the last year and finally decided to push it as an "experimental" component in the upcoming aragonUI release :).

**Pull request closed without merging**